### PR TITLE
[V2] chore: Allow for sync or async AttachDisk operation

### PR DESCRIPTION
**Pull request by edreed**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Brings the V2 driver up to parity with allowing for synchronous or asynchronous `AttachDisk` operation.

**Which issue(s) this PR fixes**:

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/provisioner/cloudprovisioner.go [hunk 1]
```diff
@@ -340,7 +340,12 @@ func (c *CloudProvisioner) PublishVolume(
 			return nil, err
 		}
 
-		lun, err = c.cloud.AttachDisk(ctx, true, diskName, volumeID, nodeName, cachingMode, disk)
+		asyncAttach := true
```
Comment (nearora-msft) [inline pkg/provisioner/cloudprovisioner.go:343]: Should we update the driver-parameters.md file to specify  that the default value of this parameter for V2 driver is true?

Comment (edreed) [reply pkg/provisioner/cloudprovisioner.go:343]: This PR is more or less so we can run the scale tests with synchronous `AttachDisk` to see its impact on the scale-out times. The full (and updated) change for this support in V1 will be brought in as part of #1216 .

Comment (nearora-msft) [reply pkg/provisioner/cloudprovisioner.go:343]: If this is just for testing, will this be reverted after #1216 is merged?

Comment (edreed) [reply pkg/provisioner/cloudprovisioner.go:343]: We'll still need the updated `cloud-provisioner-azure` brought in by this change to enable sync/async operation via configuration to maintain feature parity with the v1 driver.

**Review by nearora-msft — APPROVED** [review]

File: pkg/provisioner/cloudprovisioner.go (hunk 1 above)
Comment (edreed) [reply pkg/provisioner/cloudprovisioner.go:343]: I updated the V2 driver parameter documentation.

**Pull request merged**